bfasst/flows/xilinx_structural_error_injection_flow.py

```diff
@@ -0,0 +1,70 @@
+"""XilinxStructuralErrorInjection flow"""
+
+import random
+
+from bfasst.compare.structural import StructuralCompareTool
+from bfasst.flows.flow import Flow
+from bfasst.flows.xilinx_phys_netlist_xrev_flow import XilinxPhysNetlistXrevFlow
+from bfasst.transform.error_injector import ErrorInjector as SdnErrorInjector
+from bfasst.transform.error_injector import ErrorType
+from bfasst.types import ToolType
+
+
+class XilinxStructuralErrorInjectionFlow(Flow):
+    """XilinxStructuralErrorInjection flow"""
+
+    def get_corrupt_netlist_path(self):
+        return self.design.corrupted_netlist_path
+
+    def generate_path_to_save_corrupted_netlist(self, num_problems, err):
+        return self.design.path / f"{err}_error_{num_problems}.v"
+
+    def _run(self):
+        """Inject errors into FASM2BELS netlist and compare with Conformal"""
+
+        if "--max_dsp" not in self.flow_args[ToolType.SYNTH]:
+            self.flow_args[ToolType.SYNTH] += " --max_dsp 0"
+
+        XilinxPhysNetlistXrevFlow(self.design, self.flow_args).run()
+
+        random.seed(0)
+        injection_types = [ErrorType.BIT_FLIP, ErrorType.WIRE_SWAP]
+
+        compare_tool = StructuralCompareTool(
+            self.design,
+            self.design.phys_netlist_path,
+            self.design.reversed_netlist_path,
+            self.flow_args[ToolType.CMP],
+        )
+        error_injector = SdnErrorInjector(cwd=self.design.build_dir, design=self.design)
+
+        for err in injection_types:
+            num_problems = 0
+            num_runs = 100
+            for i in range(1, num_runs + 1):
+                # Give the injector a number for logging the run, and run the injection
+                error_injector.run_num = i
+                error_injector.inject(err)
+
+                # Reset the mappings in the compare tool, give it a number for logging the run,
+                # and run the comparison
+                compare_tool.reset_mappings()
+
+                # The compare tool should only reset its logs once per flow,
+                # not per error type, but should be unaware of what type of error is being injected
+                compare_tool.run_num = i if err == injection_types[0] else i + num_runs
+
+                if compare_tool.compare_netlists() is None:
```

Comment:
This may not be what we need, since structural.compare will raise an exception if the two netlists are not equivalent. If _no exception is raised_ by `compare_netlists()`, we've got a problem and should run the code in this if block.